packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -139,28 +139,27 @@ class TransactionService {
        * allow a redefinition until there's a call that doesn't have that
        * or the threshold is exceeded.
        */
+      let redefineType = tr.type
+      const currentTypeOrder = TRANSACTION_TYPE_ORDER.indexOf(tr.type)
+      const redefineTypeOrder = TRANSACTION_TYPE_ORDER.indexOf(type)
+
+      /**
+       * Update type based on precedence defined in TRANSACTION_TYPE_ORDER.
+       * 1. If both orders doesn't exist, we don't redefine the type.
+       * 2. If only the redefined type is not present in predefined order, that implies
+       * it's a user defined type and it is of higher precedence
+       */
+      if (currentTypeOrder >= 0 && redefineTypeOrder < currentTypeOrder) {
+        redefineType = type
+      }
       if (__DEV__) {
         this._logger.debug(
           `redefining transaction(${tr.id}, ${tr.name}, ${tr.type})`,
           'to',
-          `(${name}, ${type})`,
+          `(${name || tr.name}, ${redefineType})`,
```

Comment:
sometimes, the name is undefined so it's better to log the redefined name without any changes.